documentation/content/developmentworkflow.rst

```diff
@@ -120,3 +120,74 @@ To configure this Git pre-commit hook:
 
 The next time you try to commit your work, your pre-commit hook will run both
 shellcheck and shfmt, and warn you of any errors you may have made.
+
+Kwreview
+========
+Another way to have your code checked against ``shfmt`` and ``shellcheck`` is to
+use ``scripts/kwreview.sh``. By default, it will check your current patch (i.e.
+what has changed since the branch unstable) and print ``shellcheck``'s warnings
+and ``shfmt``'s warnings as diffs. Use ``kwreview -w`` to apply ``shfmt``'s
+changes to the files. Make sure you have ``shfmt``, ``shellcheck`` and
+``reviewdog`` installed (see :ref:`dependencies_label`).
+
+For vim users, it is possible to use ``kwreview.sh`` to populate the
+`quickfix <http://vimdoc.sourceforge.net/htmldoc/quickfix.html>`_ (or locations)
+window, making it easy to navigate the warnings and errors. As an example,
+consider adding the following lines to your ``.vimrc``::
+
+  function Kwreview(...)
+    let arg = get(a:, 1, "")
+    let &l:makeprg="scripts/kwreview.sh " . arg
+    make
+    execute ":e"
+  endfunction
+
+  command -nargs=* Kwreview :call Kwreview(<q-args>)
+
+Now you can call ``kwreview.sh`` with the vim command ``:Kwreview``
+(notice the upper case initial) and navigate the errors/warnings with
+``:copen``, ``:cn``, ``:cp``, etc.
+
+.. _dependencies_label:
+
+Dependencies
+============
+
+Installing ShellCheck
+~~~~~~~~~~~~~~~~~~~~~
+Shellcheck is available as a package for most distributions. On debian based
+distros, use ``sudo apt install shellcheck``. On Arch based distros, use ``sudo
+pacman -S shellcheck``.
+
+Installing shfmt
+~~~~~~~~~~~~~~~~
+
+To install shfmt, ``go`` must be installed. On debian based distros::
+
+  sudo apt install golang
+
+On arch based distros::
+
+  sudo pacman -S go
+
+With go installed, run the following command to install shfmt::
+
+  GO111MODULE=on go get mvdan.cc/sh/v3/cmd/shfmt
+
+More information can be found in their `github page <https://github.com/mvdan/sh>`__.
+
+Installing pre-commit
+~~~~~~~~~~~~~~~~~~~~~
+``pre-commit`` is available as a package for most distributions. On debian based
+distros, use ``sudo apt install pre-commit``. On Arch based distros, use ``sudo
+pacman -S python-pre-commit``
+
+Installing reviewdog
+~~~~~~~~~~~~~~~~~~~~
+To install reviewdog, run the following command::
+
+  curl -sfL https://raw.githubusercontent.com/reviewdog/reviewdog/master/install.sh | sh -s
```

Comment:
fwiu, if you use this method, it will install reviewdog in `$PWD/bin/`. It looks like that your script does not handle this situation. Also, even after changing your code, it does not give me any update; I tried in the `src/plugins/subsystems/drm/drm.sh` which has some shellcheck problems, but I did not get anything (:Kwreview)

Reply:
> fwiu, if you use this method, it will install reviewdog in $PWD/bin/. It looks like that your script does not handle this situation.

I changed the documentation a bit, see if it is better. 

> Also, even after changing your code, it does not give me any update; I tried in the src/plugins/subsystems/drm/drm.sh which has some shellcheck problems, but I did not get anything (:Kwreview)

By default, `kwreview` will compare the current patch with unstable, and will ignore the staging area and the working tree. I just added an option to compare against working tree (`-t`) and agains the staging area (`-g`). Tell me if that was not the problem.